### PR TITLE
UGENE-7365. Move columns from MSA to a new document

### PR DESCRIPTION
**Pull request by yalgaer**
To move rows easily between alignment objects we need an easy way to create a new document, which plays the role of exclude list. This change adds a menu option to move sequences to a new file.
https://ugene.dev/tracker/browse/UGENE-7365

I also had to move export task from the export plugin to corelibs (that was an old TODO), document the changed code and fix warnings. The GUI test will be added tomorrow, first I want to see this changed does not break anything.

### CODE REVIEW

File: src/corelibs/U2Core/src/tasks/ExportTasks.cpp [hunk 1]
```diff
@@ -223,31 +189,25 @@ void ExportMSA2MSATask::run() {
...
 
 void ExportDNAChromatogramTask::prepare() {
-    Document *d = cObj->getDocument();
-    assert(d != nullptr);
-    if (d == nullptr) {
-        stateInfo.setError("Chromatogram object document is not found!");
-        return;
-    }
+    Document *d = chromaObject->getDocument();
+    SAFE_POINT_EXT(d != nullptr, stateInfo.setError("Chromatogram object document is not found!"), );
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Core/src/tasks/ExportTasks.cpp]: `L10N::nullPointerError` could be used for this type of errors

File: src/corelibs/U2Core/src/tasks/ExportTasks.cpp [hunk 2]
```diff
@@ -223,31 +189,25 @@ void ExportMSA2MSATask::run() {
...
-    assert(relatedObjs.count() == 1);
-    if (relatedObjs.count() != 1) {
-        stateInfo.setError("Sequence related to chromatogram is not found!");
-    }
     QString seqObjName = relatedObjs.first().ref.objName;
 
     GObject *resObj = d->findGObjectByName(seqObjName);
     U2SequenceObject *sObj = qobject_cast<U2SequenceObject *>(resObj);
-    assert(sObj != nullptr);
+    SAFE_POINT_EXT(sObj != nullptr, stateInfo.setError(L10N::internalError()), );
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Core/src/tasks/ExportTasks.cpp]: Maybe `L10N::nullPointerError`

Comment (yalgaer) [reply src/corelibs/U2Core/src/tasks/ExportTasks.cpp]: Done

File: src/corelibs/U2Core/src/tasks/ExportTasks.cpp (hunk 1 above)
Comment (yalgaer) [reply src/corelibs/U2Core/src/tasks/ExportTasks.cpp]: Done

File: src/corelibs/U2View/src/ov_msa/move_to_object/MoveToObjectMaController.cpp [hunk 3]
```diff
@@ -127,4 +129,68 @@ void MoveToObjectMaController::buildMenu(GObjectView *, QMenu *menu, const QStri
...
+    auto removeRowsTask = new RemoveRowsFromMaObjectTask(editor, rowIdsToRemove);
+    auto task = new MultiTask(tr("Export alignment rows to a new file"), {createNewMsaTask, removeRowsTask});
+    AppContext::getTaskScheduler()->registerTopLevelTask(task);
+}
+
+/************************************************************************/
+/* RemoveRowsFromMsaObject */
+/************************************************************************/
+RemoveRowsFromMaObjectTask::RemoveRowsFromMaObjectTask(MaEditor *_maEditor, const QList<qint64> &_rowIds)
+    : Task(tr("Remove rows from alignment"), TaskFlag_RunInMainThread), maEditor(_maEditor), rowIds(_rowIds) {
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2View/src/ov_msa/move_to_object/MoveToObjectMaController.cpp:188]: Why do you need to run it in the main thread? The database updating (`U2UseCommonUserModStep`) could be a bit time-consuming

**Review by DmitriiSukhomlinov — CHANGES_REQUESTED** [review]
Do you want to add several counters to this feature?

File: src/corelibs/U2View/src/ov_msa/move_to_object/MoveToObjectMaController.cpp (hunk 3 above)
Comment (yalgaer) [reply src/corelibs/U2View/src/ov_msa/move_to_object/MoveToObjectMaController.cpp:188]: This task performs UI ops.
Is there any place in our code where U2UseCommonUserModStep are run in non-main thread?

**Comment by yalgaer** [comment]
> Do you want to add several counters to this feature?

Added

**Review by DmitriiSukhomlinov — APPROVED** [review]

**Pull request merged**